account/verification_code.go

```diff
@@ -0,0 +1,189 @@
+type VerificationCodeRepository interface {
+	repository.Exister
+	Load(ctx context.Context, id uuid.UUID) (*VerificationCode, error)
+	LoadByCode(ctx context.Context, code string) ([]VerificationCode, error)
+	Create(ctx context.Context, VerificationCode *VerificationCode) error
+	Save(ctx context.Context, VerificationCode *VerificationCode) error
+	Delete(ctx context.Context, id uuid.UUID) error
+	Query(funcs ...func(*gorm.DB) *gorm.DB) ([]VerificationCode, error)
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *GormVerificationCodeRepository) TableName() string {
+	return "verification_codes"
+}
+
+// CRUD Functions
+
+// Load returns a single VerificationCode as a Database Model
+// This is more for use internally, and probably not what you want in  your controllers
+func (m *GormVerificationCodeRepository) Load(ctx context.Context, id uuid.UUID) (*VerificationCode, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "VerificationCode", "load"}, time.Now())
+
+	var native VerificationCode
+	err := m.db.Table(m.TableName()).Where("id = ?", id).Find(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil, errors.NewNotFoundError("verification_code_id", id.String())
+	}
+
+	return &native, errs.WithStack(err)
+}
+
+// LoadByCode loads result by filtering with respect to the verificate code.
+func (m *GormVerificationCodeRepository) LoadByCode(ctx context.Context, code string) ([]VerificationCode, error) {
+	return m.Query(VerificationCodeWithUser(), VerificationCodeFilterByCode(code)) // maybe load with user?
+}
+
+// CheckExists returns nil if the given ID exists otherwise returns an error
+func (m *GormVerificationCodeRepository) CheckExists(ctx context.Context, id string) error {
+	defer goa.MeasureSince([]string{"goa", "db", "VerificationCode", "exists"}, time.Now())
+	return repository.CheckHardDeletableExists(ctx, m.db, m.TableName(), id)
+}
+
+// Create creates a new record.
+func (m *GormVerificationCodeRepository) Create(ctx context.Context, model *VerificationCode) error {
+	defer goa.MeasureSince([]string{"goa", "db", "VerificationCode", "create"}, time.Now())
+	if model.ID == uuid.Nil {
+		model.ID = uuid.NewV4()
+	}
+	err := m.db.Create(model).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"verification_code_id": model.ID,
+			"err": err,
+		}, "unable to create the verification_code")
+		return errs.WithStack(err)
+	}
+	log.Info(ctx, map[string]interface{}{
+		"verification_code_id": model.ID,
+	}, "verification_code created!")
+	return nil
+}
+
+// Save modifies a single record.
+func (m *GormVerificationCodeRepository) Save(ctx context.Context, model *VerificationCode) error {
+	defer goa.MeasureSince([]string{"goa", "db", "VerificationCode", "save"}, time.Now())
+
+	obj, err := m.Load(ctx, model.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"verification_code_id": model.ID,
+			"ctx": ctx,
```

Comment:
No need to add ctx explicitly. It will be added by log.Error()

Reply:
19e25e8 done.